programs/us/us.cpp

```diff
@@ -644,82 +723,223 @@ void US_Win::closeEvent( QCloseEvent* e )
 
 void US_Win::splash( void )
 {
-  const int y =           menuBar  ()->size().rheight();
-  const int height = 532 - y - statusBar()->size().rheight();
+  const int y = menuBar()->size().rheight();
+  const int h = 532 - y - statusBar()->size().rheight();
   const int w = 710;
 
   bigframe = new QLabel( this );
   bigframe->setFrameStyle        ( QFrame::Box | QFrame::Raised);
   bigframe->setPalette           ( US_GuiSettings::frameColor() );
-  bigframe->setGeometry          ( 0, y, w, height );
+  bigframe->setGeometry          ( 0, y, w, h );
   bigframe->setAutoFillBackground( true );
 
-  splash_shadow = new QLabel( this );
-  splash_shadow->setGeometry( (unsigned int)( ( w / 2 ) - 210 ) , 130, 460, 276 );
-  splash_shadow->setPalette( QPalette( Qt::black, Qt::cyan ) );
-  splash_shadow->setAutoFillBackground ( true );
+  if (!bigframe) bigframe = new QLabel(this);
+    Q_ASSERT(bigframe);
+    bigframe->setGeometry(0, y, w, h);
+    bigframe->setFrameStyle(QFrame::NoFrame);
+    bigframe->setAutoFillBackground(false); // image covers all
+
+    if (splash_shadow) { splash_shadow->hide(); splash_shadow->deleteLater(); splash_shadow = nullptr; }
+
+    logo(w);
+}
+
+static QPixmap makeSplashCanvas(const QPixmap& rawSplash,
+                                const QString& versionLine,
+                                const QString& builtLine,
+                                const QStringList& authors,
+                                int maxW, int maxH, int marginPx)
+{
+    const int canvasW = qMax(1, maxW);
+    const int canvasH = qMax(1, maxH);
+    const int margin  = qMax(4, marginPx);
+
+    // HiDPI-safe canvas
+    const qreal dpr = qApp->primaryScreen()->devicePixelRatio();
```

Comment:
```suggestion
    const qreal dpr = QGuiApplication::primaryScreen()->devicePixelRatio();
```
qApp's return time can be QApplication or QGuiApplication